R/OptimInstance.R

```diff
@@ -49,10 +49,15 @@ OptimInstance = R6Class("OptimInstance",
 
       assert_choice(keep_evals, c("all", "best"))
       self$objective = assert_r6(objective, "Objective")
-      self$search_space = if (is.null(search_space)) {
+
+      self$search_space = if (is.null(search_space) && self$objective$domain$search_space()$length == 0) {
         self$objective$domain
-      } else {
+      } else if (is.null(search_space) && self$objective$domain$search_space()$length > 0) {
```

Comment:
I think it is better to get the `self$objective$domain$search_space()` only once and save it in a temp variable, since the `search_space()` call can be a bit slow.